src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -62,12 +62,18 @@ class MapViewVisualization extends Visualization {
         mapElem.setAttributeNS(null,"style","width:" + width + "px;height: " + height + "px;");
         this.dom.appendChild(mapElem);
 
-        const parsedData = eval('('+data+')' );
+        const parsedData = eval('('+data+')' ) || JSON.parse(data);
```

Comment:
ok, this is scarry. What guarantees you have that data would not contain some kind of magical string which would not cause harm here? Imagine something like that - In enso, we query HTTP endpoint, and we get string back, from third party server. Then this string is sent to GUI (lets assume that by accident or somehow it is not converted to JSON) earlier and it is evaluated here. This is a BIG security concern. We should never ever use eval unless we have to. Why just using JSON parse would not be enough here?